### ISSUE TITLE
Evil-snipe-mode breaks deer-mode

### ISSUE THREAD

**Issue by martinzk** [syl20bnr/spacemacs#7645]
#### Description :octocat:
Keybindings in deer mode are overwritten by evil-snipe layer.

I have tried with dired and ranger hooks to call turn-off-evil-snipe-mode and turn-off-evil-snipe-override-mode but evil-snipe is still loaded in deer.

#### Reproduction guide :beetle:
- Start Emacs
- SPC a d
- f files calls evil-snipe-f
- ; calls evil-snipe-repeat

*Expected behaviour:* :heart: :smile:
Should call the ranger/deer bindings.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 24.5.2
- Spacemacs: 0.200.4
- Spacemacs branch: develop (rev. a6554a5)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm
 (auto-completion :variables auto-completion-private-snippets-directory "~/.spacemacs.d/snippets/" auto-completion-enable-snippets-in-popup t)
 bibtex better-defaults emacs-lisp
 (evil-snipe :variables evil-snipe-enable-alternate-f-and-t-behaviors t)
 fasd git github gtags ibuffer
 (latex :variables latex-enable-auto-fill 'nil)
 markdown nixos org python
 (ranger :variables ranger-override-dired t)
 semantic
 (shell :variables shell-default-height 30 shell-default-position 'bottom shell-enable-smart-eshell t)
 shell-scripts spell-checking syntax-checking systemd typography
 (version-control :variables version-control-diff-tool 'diff-hl))
```

**Comment by TheBB** [comment]
I'm not sure how to fix this. It's turned off in deer, as it should be, but then turned back on again as part of `post-command-hook` due to being a globalized minor mode.

**Comment by TheBB** [comment]
https://github.com/hlissner/evil-snipe/pull/45

**Comment by TheBB** [comment]
Actually this problem runs deeper, I'll have a look but it will not be easy to fix I think.

**Comment by braham-snyder** [comment]
This bug is pretty odd to me... I'm seeing, on develop (haven't looked at master), `evil-snipe` commands shadowing major mode commands despite `evil-snipe-local-mode` being `nil`. I can't repro with `SPC q D`.

I seem to encounter the same issue with `evil-snipe-s` in `undo-tree-visualizer` (which should be `undo-tree-visualizer-selection-mode`)

in ranger-mode, I think I've cir

**Comment by braham-snyder** [comment]
Is this the same bug causing `undo-tree-visualizer-select` to be shadowed by `evil-snipe-s` regardless of `evil-snipe-local-mode` being `nil`? Seeing these issues on both master and develop.

Regarding repro with ranger: note that enabling `evil-snipe-override-mode` from a `deer` buffer doesn't immediately shadow `ranger-travel` (`f`)--shadowing does occur after navigating to a different folder or reopening deer.

With ranger, I've circumvented the issue by using normal state instead of motion state (which doesn't work in `undo-tree-mode`)--in ranger.el, I just changed:
```
         (evil-set-initial-state 'ranger-mode 'motion)
         (evil-make-overriding-map ranger-mode-map 'motion)
```
to
```
         (evil-set-initial-state 'ranger-mode 'normal)
         (evil-make-overriding-map ranger-mode-map 'normal)
```

I'm sort of confused that it doesn't seem like many people are running into this issue... Do not many people use `evil-snipe-override`, or something? Is there a simpler workaround?
